jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java

```diff
@@ -614,15 +565,42 @@ public boolean abort(HttpExchange exchange, Throwable failure)
         }
     }
 
+    private boolean updateResponseState(ResponseState from1, ResponseState from2, ResponseState to)
```

Comment:
I'm not fond of this method's signature, what do you think of that one instead?
```java
private boolean updateResponseState(EnumSet<ResponseState> from, ResponseState to)
```
that would make the callers look like this:
```java
if (updateResponseState(EnumSet.of(ResponseState.HEADERS, ResponseState.CONTENT), ResponseState.TRANSIENT))
```

Reply:
It allocates :smiley: